src/menus/menus.service.ts

```diff
@@ -433,21 +435,21 @@ export class MenusService {
         where: { menuId, id: revisionId },
       });
 
-      const menuItems = await menu.items;
+      const snapshotItems = await revision.snapshot.items;
 
-      const getChildren = (menuItems: MenuItem[], item: MenuItem) => {
-        const children = menuItems.filter((i) => i.parentId === item.id);
+      const getChildren = (snapshotItems: MenuItem[], item: MenuItem) => {
+        const children = snapshotItems.filter((i) => i.parentId === item.id);
         return children.map((c) => ({
           ...c,
-          children: getChildren(menuItems, c),
+          children: getChildren(snapshotItems, c),
         }));
       };
 
-      const items = menuItems
+      const items = snapshotItems
         .filter((i) => !i.parentId)
         .map((i: MenuItem) => ({
           ...i,
-          children: getChildren(menuItems, i),
+          children: getChildren(snapshotItems, i),
```

Comment:
Here is the fix for the mysterious bug that happened when publishing the Menu. Fixes https://github.com/bancodobrasil/featws-agile/issues/126